### ISSUE TITLE
[Enhancement] Component Request: Standardized Detail List with Header for Information Display

### ISSUE THREAD

**Issue by Sk0vgaard** [kirbydesign/designsystem#3701]
- [X] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the solution you'd like
![image](https://github.com/user-attachments/assets/084da003-27d9-4d91-957f-f43a04d8b984)

We need a component to display a list of details with a header, similar to the example provided in the attached screenshot.

The desired component should:
- Present a header at the top, representing what the list of details is about (e.g., "Details").
- Show key-value pairs for displaying information succinctly.
- Allow for multiple fields, potentially with options for hiding sensitive data or partial obfuscation (e.g., for card numbers).
- Maintain a clean and responsive design, aligning with Kirby’s visual style.
- Possiblible to use either KeyValueListItem[] or create multiple key-value-list-item's in the html

## Have you considered any alternatives?
Currently, each team has created custom implementations that look somewhat similar but lack uniformity. Using a Kirby component will:
- Reduce the need for redundant code.
- Ensure design consistency.
- Improve maintainability by having a single, reusable component across teams.
- An alternative would be to enhance the existing components with extensions, but this would likely add unnecessary complexity to the codebase and dilute design uniformity.

## Are there any additional context?
This component is already in use within the monolithic architecture of an existing application, where five teams currently rely on it. Given that multiple other teams have created their own versions of a similar component, consolidating this into a Kirby component would provide consistency across applications.
Currently selector used in monolith is: "_drb-key-value-list_" & "_drb-key-value-list-item_"
It should reuse the "_kirby-section-header_" selector as the header.

<hr />

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#the-process-of-contributing) correctly.

### Refinement

- [x] Request that the issue is [UX refined](https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#ux-refinement); do not proceed until this is done.
- [X] Request that the issue is [tech refined](https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#tech-refinement); do not proceed until this is done.

### Implementation 
The contributor who wants to implement this issue should: 

- [ ] Make sure you have read: "[Before you get coding](../CONTRIBUTING.md/#before-you-get-coding)".
- [ ] Signal to others you are working on the issue by assigning yourself.
- [ ] Create a branch from the [develop branch](https://github.com/kirbydesign/designsystem/tree/develop) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [ ] Publish a WIP implementation to Github as a draft PR and ask for feedback. 
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Update the [cookbook](https://cookbook.kirby.design) with examples and showcases.

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request. If you created a draft PR during implementation you can just mark that as "ready for review".

**Comment by RasmusKjeldgaard** [comment]
Closing this issue automatically because it has been open for more than 6 months and is not currently in progress. If this is still relevant, please feel free to reopen it or open a new issue with updated details. Thank you!
